### ISSUE TITLE
[Cloud Firestrore Plugin] Snapshot.hasData not working

### ISSUE THREAD

**Issue by kroikie** [FirebaseExtended/flutterfire#1201]
```
snapshot.hasData -> true
snapshot.data == false -> false
snapshot.data['uid'] -> error. [] called on null
```
Here is the repro video:
https://youtu.be/coMa7gydwLk?t=252

Temporary workaround: 
```
                  bool isUserNull = false;
                  try {
                    var userUID = snapshot.data['uid'];
                  } catch (err) {
                    print('USER DOCCUMENT MIGHT BE NULL: $err');
                    isUserNull = true;
                  }
```

**Comment by kroikie** [comment]
@raveesh-me

The issue at https://github.com/flutter/flutter/issues/39097 has been closed and moved here. Future collaboration on this issue will be done here.

**Comment by Salakar** [comment]
Hey, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) (via the `snapshot.exists` property) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
